modules/ballerina-core/src/main/java/org/ballerinalang/model/values/BMap.java

```diff
@@ -51,7 +52,12 @@
     private final Lock writeLock = lock.writeLock();
 
     public BMap() {
-        map =  new LinkedHashMap<>();
+        map = new LinkedHashMap<>();
+    }
+
+    public BMap(BType type) {
```

Comment:
It is better we can constraint this type BMapType. Otherwise, people can misuse this constructor.

Reply:
+1